src/lib/party.ts

```diff
@@ -0,0 +1,784 @@
+import { CircuitValue } from './circuit_value';
+import { Group, Field, Bool, VerificationKey } from '../snarky';
+import { PrivateKey, PublicKey } from './signature';
+import { UInt64, UInt32, Int64 } from './int';
+import * as Mina from './mina';
+
+export type Amount = UInt64;
+export const Amount = UInt64;
+export type Balance = UInt64;
+export const Balance = UInt64;
+export type Fee = UInt64;
+export const Fee = UInt64;
+export type GlobalSlot = UInt32;
+export const GlobalSlot = UInt32;
+export type SignedAmount = Int64;
+export const SignedAmount = Int64;
```

Comment:
I'm not a fan of having many names for the same types / classes - that just adds complexity IMO, our brain can't remember so many names

Reply:
I disagree, the types become nice documentation. Ie. If I see an arbitrary UInt64, is it an Amount or a Fee?  It would be better to add more type-safety beyond aliases so we don't mix these up https://basarat.gitbook.io/typescript/main-1/nominaltyping could be a good approach, but maybe it's too hacky?

Reply:
Hm, interesting take @ documentation. I want to defend my position a bit (even if I'm perfectly fine with leaving the code as it is)

In my mind, variable names, function arguments and object properties should already all have descriptive names, so a descriptive name of the accompanying type doesn't add value and brings other drawbacks

For example, say we want to interact with a function that has the signature
`function(accountCreationFee: UInt64)`
Then we'll be happy, because we know that we're passing in a fee, and we also know how to create that fee - by instantiating a `UInt64`, which is a type we use all the time in snarkyJS so we probably did that a couple of times or have code to see how it is done. We also have a mental model already for what a `UInt64` is under the hood.

On the other hand, if we see this signature:
`function(accountCreationFee: Fee)`
Then the "fee" information is redundant, and we probably need to search for a while to figure out how to create such a `Fee` from the `number` we probably have. Also it may be frustrating that we don't understand what a `Fee` created from a number even *is*, so we'll have to settle for a shallower understanding of our program than in the example above.

I think this example can be generalized to basically any interaction with any type

Reply:
I'm also concerned about the namespace pollution this causes. "Minimal API surface area" is one of the very few programming paradigms that I deeply believe in. party.ts has _26 exports_